website/addons/mendeley/views.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+
+from flask import request
+
+from framework.auth.decorators import must_be_logged_in
+
+from website.project.decorators import (
+    must_be_contributor_or_public,
+    must_have_permission,
+    must_not_be_registration,
+    must_have_addon,
+)
+
+from .provider import MendeleyCitationsProvider
+
+
+@must_be_logged_in
+def list_mendeley_accounts_user(auth):
+    """ Returns the list of all of the current user's authorized Mendeley accounts """
+
+    provider = MendeleyCitationsProvider()
+    return provider.user_accounts(auth.user)
+
+
+@must_have_permission('read')
+@must_have_addon('mendeley', 'node')
+def mendeley_get_config(auth, node_addon, **kwargs):
+    """ Serialize node addon settings and relevant urls
+    (see serialize_settings/serialize_urls)
+    """
+
+    provider = MendeleyCitationsProvider()
+    return provider.serialize_settings(node_addon, auth.user)
+
+@must_have_permission('write')
+@must_have_addon('mendeley', 'node')
+@must_not_be_registration
+def mendeley_set_config(auth, node_addon, **kwargs):
+    """ Updates MendeleyNodeSettings based on submitted account and folder information """
+
+    provider = MendeleyCitationsProvider()
+    args = request.get_json()
+    #external_account_id = args.get('external_account_id')
+    external_list_id = args.get('external_list_id')
+    provider.set_config(
+        node_addon,
+        auth.user,
+        external_list_id,
+    )
+    return {}
```

Comment:
Views should always return some feedback to the client. It might make sense to return the serialize settings here.

I will make this change when I merge.